File: c++/desafio2.cpp
#include <iostream>
#include <vector>

using namespace std;

const int N = 289;

bool esEscalonado(int number){
    string numStr = to_string(number);
    int numDigits = numStr.length();
    
    for (int i = 1; i < numDigits; i++) {
        if (numStr[i] <= numStr[i - 1]) {
            return false;
        }
    }
    
    return true;
}

int contarEscalonados(int num){
    int res = 0;
    vector<int> escalonados = {};
    
    for(int i = 10; i <= num; i++){
        if(esEscalonado(i)){
            escalonados.push_back(i);
            res++;
        }
    }
    
    for (int num : escalonados) {
        cout << num << "\n";
    };

    return res;
}

int main(){
    cout << contarEscalonados(N) << "\n";

    return 0;
}

File: js/desafio2.js
function esEscalonado(number) {
    const digitos = String(number).split('').map(Number); //para convertir numero en array
    for (let i = 1; i < digitos.length; i++) {
        if (digitos[i] <= digitos[i - 1]) { //si digito siguiente es menor al digito anterior es menor no es escalonado
            return false;
        }
    }
    return true;
}

function contarEscalonados(N) {
    let cont = 0;
    let escalonados = [];
    for (let i = 10; i <= N; i++) {
        if (esEscalonado(i)) {
            cont++;
            escalonados.push(i);
        }
    }
    
    console.log("Cantidad de nros escalonados: " + cont);
    return "Nros escalonados: " + escalonados;
} 

console.log(contarEscalonados(19999));